zaza/openstack/charm_tests/ceph/tests.py

```diff
@@ -1748,3 +1748,38 @@ def test_persistent_config(self):
             )
             data = json.loads(result['Stdout'])
             assert data['loglevel'] == 2
+
+
+class CephMonKeyRotationTests(test_utils.BaseCharmTest):
+    """Tests for the rotate-key action."""
+
+    def _get_mgrs_keys(self, unit):
+        cmd = 'sudo ceph auth ls'
+        result = zaza_model.run_on_unit(unit, cmd)
+        # Don't use json formatting, as it's buggy upstream.
+        data = result['Stdout'].split()
+        ret = set()
+
+        for ix, line in enumerate(data):
+            # Structure:
+            # mgr.XXXX
+            # key:
+            # key contents
+            # That's why we need to move 2 positions ahead.
+            if line.startswith('mgr.'):
+                ret.add(data[ix + 2])
+        return ret
+
+    def test_mgr_key_rotate(self):
+        """Test that rotating the manager key actually changes it."""
+        unit = 'ceph-mon/0'
+        old_keys = self._get_mgrs_keys(unit)
+
+        action_obj = zaza_model.run_action(
+            unit_name=unit,
+            action_name='rotate-key',
+            action_params={'entity': 'mgr'}
+        )
+        zaza_utils.assertActionRanOK(action_obj)
+        new_keys = self._get_mgrs_keys(unit)
+        self.assertNotEqual(old_keys, new_keys)
```

Comment:
Believe this would succeed if just one of the keys were changed -- suggest to check that actually each key is changed

Reply:
This only rotates the key for the manager on the ceph-mon unit specified. Getting the manager for a mon is rather complicated, since managers don't have stable names (they change on upgrade, for instance).
I can harden it a bit by asserting that only one key changed, though.